bloom/infra/spire_service.py

```diff
@@ -0,0 +1,155 @@
+from logging import getLogger
+from gql import Client
+from gql.transport.requests import RequestsHTTPTransport
+from requests import exceptions
+from collections.abc import MutableMapping
+
+from bloom.domain.vessel import Vessel
+from bloom.infra.spire_api_paging import Paging
+
+logger = getLogger()
+
+class SpireService:
+    def __init__(self) -> None:
+        self.transport = RequestsHTTPTransport(
+            url='https://api.spire.com/graphql',
+            # headers={'Authorization': f"Bearer {os.getenv('SPIRE_TOKEN')}"},
+            headers={'Authorization': 'Bearer U2hWhOiuDfAiVo8g07IIkMdzrDZtPGps'},
+            verify=True,
+            retries=3,
+            timeout=30)
+
+    def create_client(self):
+        try:
+            client = Client(transport=self.transport, fetch_schema_from_transport=True)
+        except exceptions.ConnectTimeout as e:
+            logger.error(e)
+            raise
+        return client
+    
+    def create_query_string(self, vessels: list[Vessel]):
+        imo_list = [vessel["IMO"] for vessel in vessels]
+        gql_formatted_imo_list = '\n'.join(map(str, imo_list))
+
+        return f"""
+        query {{
+            vessels(
+                imo: [
+                    9175834
+                    227302000
+                    244309000
+                    9204556
+                    8028412
+                    244070881
+                    8707537
+                    8209171
+                    9074951
+                    8716928
+                    9126364
+                    8707446
+                    9182801
+                    8301187
+                    9690688
+                    8918318
+                    8707745
+                    8224406
+                    9828936
+                    9187306
+                    9249556
+                    9249568
+                    8901913
```

Comment:
This will be replaced by `gql_formatted_imo_list`

I think that it's also possible to format gql query differently using params, here I use string interpolation because it's easier, and I don't see a real advantage of using gql params.